connectors/sources/redis.py

```diff
@@ -0,0 +1,283 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import json
+from contextlib import asynccontextmanager
+from enum import Enum
+
+import redis.asyncio as redis
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource, ConfigurableFieldValueError
+from connectors.utils import hash_id, iso_utc
+
+PAGE_SIZE = 1000
+MAX_POOL_SIZE = 10
+
+
+class KeyType(Enum):
+    HASH = "hash"
+    STREAM = "stream"
+    LIST = "list"
+    SET = "set"
+    ZSET = "zset"
+    JSON = "ReJSON-RL"
+
+
+class RedisClient:
+    """Redis client to handle method calls made to Redis"""
+
+    def __init__(self, configuration):
+        self.configuration = configuration
+        self._logger = logger
+        self.host = self.configuration["host"]
+        self.port = self.configuration["port"]
+        self.database = self.configuration["database"]
+        self.username = self.configuration["username"]
+        self.password = self.configuration["password"]
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+
+    @asynccontextmanager
+    async def client(self):
+        pool = redis.ConnectionPool.from_url(
+            f"redis://{self.username}:{self.password}@{self.host}:{self.port}",
+            decode_responses=True,
+            max_connections=MAX_POOL_SIZE,
+        )
+        yield redis.Redis(connection_pool=pool)
```

Comment:
Should you do the cleanup in the `finally`?

In the official [example](https://redis.readthedocs.io/en/stable/examples/asyncio_examples.html), if you create client with `from_pool` class method, you will only need to call `await client.close()` to close it.

But if you use `redis.Redis(connection_pool=pool)`, you need to explicitly close the client and the pool.